src/marqo/s2_inference/inference_cache/inference_cache.py

```diff
@@ -0,0 +1,77 @@
+from typing import List, Optional
+
+from readerwriterlock import rwlock
+
+from marqo.api.exceptions import EnvVarError
+from marqo.s2_inference.inference_cache.abstract_cache import MarqoAbstractCache
+from marqo.s2_inference.inference_cache.marqo_lfu_cache import MarqoLFUCache
+from marqo.s2_inference.inference_cache.marqo_lru_cache import MarqoLRUCache
+
+
+class InferenceCache:
+    """Inference cache class that wraps MarqoAbstractCache and LRUCache with a read-write lock.
+
+    The key is a string consisting of model_cache_key and content to identify the cache.
+    The value is a list of floats representing the embeddings.
+    """
+    _CACHE_TYPE_MAPPINGS = {
+        "LFU": MarqoLFUCache,
+        "LRU": MarqoLRUCache
+    }
+
+    def __init__(self, cache_size: int, cache_type: str = "LRU"):
```

Comment:
Rather than cache type it would make more sense to get an instance of `MarqoAbstractCache` here